go/cmd/dolt/commands/fetch.go

```diff
@@ -179,19 +186,32 @@ func fetchRefSpecs(ctx context.Context, dEnv *env.DoltEnv, rem env.Remote, refSp
 			remoteTrackRef := rs.DestRef(branchRef)
 
 			if remoteTrackRef != nil {
-				verr := fetchRemoteBranch(ctx, dEnv, rem, srcDB, dEnv.DoltDB, branchRef, remoteTrackRef)
+				srcDBCommit, verr := fetchRemoteBranch(ctx, dEnv, rem, srcDB, dEnv.DoltDB, branchRef, remoteTrackRef)
 
 				if verr != nil {
 					return verr
 				}
+
+				if forceFetch {
+					err = dEnv.DoltDB.DeleteBranch(ctx, remoteTrackRef)
+					if err != nil {
+						return errhand.BuildDError("error: fetch failed").AddCause(err).Build()
+					}
+				}
+
+				err =  dEnv.DoltDB.FastForward(ctx, remoteTrackRef, srcDBCommit)
```

Comment:
I know you didn't cause this...issue...but reading through the code, I think before we call this method, we need to call `dEnv.DoltDB.CanFastForward(ctx, remoteTrackRef, srcDBCommit)`

The `FastForward` method itself doesn't actually implement the correctness check. But it does seem to go through some code that has a quick sanity check that will cause it to fail for a non-FF merge in some cases...